internal/onepassword/model/item.go

```diff
@@ -0,0 +1,283 @@
+package model
+
+import (
+	"context"
+	"fmt"
+
+	connect "github.com/1Password/connect-sdk-go/onepassword"
+	sdk "github.com/1password/onepassword-sdk-go"
+)
+
+type Item struct {
+	ID       string
+	Title    string
+	VaultID  string
+	Category connect.ItemCategory
+	Version  int
+	Tags     []string
+	URLs     []ItemURL
+	Sections []*ItemSection
+	Fields   []*ItemField
+	Files    []*ItemFile
```

Comment:
I'd go here without pointers, as this will bring the following benefits:
- Fewer allocations & less GC work. []T stores the elements contiguously in one backing array. []*T needs one allocation per element, which increases GC pressure and pointer chasing.
- Better cache locality. Contiguous structs read/write faster than following pointers all over the heap.
- Simpler semantics. No nil elements to check; you just get zero-values. APIs are easier to use and reason about.
- Copy cost is often negligible. Slices are small headers; passing []T around does not copy the whole array. Even when elements move during append, copying small/medium structs is cheap.